src/provider/rainbowKit/RainbowKitAccount.ts

```diff
@@ -54,18 +54,28 @@ class RainbowKitAccount extends Eip6963Provider {
         gas: authInfo.fee?.gasLimit.toString() ?? "0",
       }
       const aminoMsgs = msgs.map(msg => AminoTypesMap.toAmino(msg))
-      const sig = await rainbowKit.signEip712(
+      const { sig, signedDoc } = await rainbowKit.signEip712(
```

Comment:
oh i think u regenerated the signdoc here, to be consistent can u do the same for metamask sign direct